### PR TITLE
Reject the modification command if multiple flow scope instances are active 

### LINKED ISSUE #10008: Reject the modification command if multiple flow scope instances are active but no ancestor is selected
## Description

The modification command can contain activation instructions for different elements. If the element is inside a flow scope (i.e. a subprocess) and there is an active instance of this flow scope then the new element is added to this flow scope instance (#9642). 

If multiple instances of this flow scope are active then the engine can't decide and the ancestor element instance must be selected (#9646).    

If at least one flow scope of an element has more than one active instance and no ancestor is selected then the command should be rejected. 

Blocked by 
- #9642

### PR DESCRIPTION
**Pull request by saig0**
## Description

Reject the modification command if a flow scope of an activation element has more than one instance. The engine can't decide in which flow scope instance the element should be created.

The verification is implemented by throwing an exception, in the same way when we fail to create the event subscriptions of a flow scope. We could implement it like the other validations in the processor but we would need to duplicate the logic to walk through the flow scopes from top to down.

## Related issues

closes #10008 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java [hunk 1]
```diff
@@ -81,6 +82,12 @@
       with a scope element that doesn't belong to the activating element's flow scope. \
       These variables should be set before or after the modification.""";
 
+  private static final String ERROR_MESSAGE_MORE_THAN_ONE_FLOW_SCOPE_INSTANCE =
+      """
+      Expected to modify instance of process '%s' but it contains one or more activate instructions \
+      for an element that has a flow scope with more than one active instances. Can't decide in \
+      which instance of the flow scope the element should be activated.""";
```
Comment (remcowesterhoud) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java:89]: 🔧 I think we can make this message more descriptive and help users with what actions they should take to fix the problem. Maybe something like this:
```
Expected to modify instance of process '%s' but it contains one or more activate instructions for an element that has a flow scope with more than one active instance: '<activation element id>'. Please specify an ancestor element instance key for this activate instruction.
```

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java [hunk 2]
```diff
@@ -81,6 +82,12 @@
       with a scope element that doesn't belong to the activating element's flow scope. \
       These variables should be set before or after the modification.""";
 
+  private static final String ERROR_MESSAGE_MORE_THAN_ONE_FLOW_SCOPE_INSTANCE =
+      """
+      Expected to modify instance of process '%s' but it contains one or more activate instructions \
+      for an element that has a flow scope with more than one active instances. Can't decide in \
```
Comment (remcowesterhoud) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java:88]:
```suggestion
      for an element that has a flow scope with more than one active instance. Can't decide in \
```

🔧 I believe this should be singular.

**Review by remcowesterhoud — APPROVED** [review]
Thanks @saig0! 🏅 

I have some small comments, please have a look before merging.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java (hunk 1 above)
Comment (saig0) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java:89]: Good point. I will adjust the message.

But I will not add the part with the ancestor (`Please specify an ancestor element instance key for this activate instruction.`) until the feature (#9646) is implemented.

**Comment by saig0** [comment]
bors merge

**Pull request merged**